internal/pkg/service/buffer/storage/level/local/writer/csv/csv.go

```diff
@@ -120,15 +120,15 @@ func NewWriter(b *base.Writer) (w *Writer, err error) {
 }
 
 func (w *Writer) WriteRow(values []any) error {
+	// Block Close method
+	w.writeWg.Add(1)
+	defer w.writeWg.Done()
+
 	// Check if the writer is closed
 	if err := w.ctx.Err(); err != nil {
 		return errors.Errorf(`CSV writer is closed: %w`, err)
 	}
 
-	// Block Close method
-	w.writeWg.Add(1)
-	defer w.writeWg.Done()
-
```

Comment:
Vsimol som si chybu, toto ma byt naopak, inak sa moze teoreticky stat, ze nejaky write bude bezat aj po `Close`.

Neda sa to nasimulovat v teste, pretoze `Close` trva vzdy dlhsie ako tento uvod `WriteRow` metody.

---------------------------

Mozny scenar:
```
WriteRow: if ctc.Err != nil -> false
Close: w.cancel
Close: w.base.Close
Close: w.writeWg.Wait (not blocked) <<<<<<<<<<
WriteRow: w.writeWg.Add ...
```

Teraz:
```
WriteRow: w.writeWg.Add ...
WriteRow: if ctc.Err != nil -> false
Close: w.cancel
Close: w.base.Close
Close: w.writeWg.Wait (blocked) <<<<<<<<<<
``` 

Alebo

```
WriteRow: w.writeWg.Add ...
Close: w.cancel
WriteRow: if ctc.Err != nil -> true -> return <<<<<<<<<<
```